File: src/network/packet/impl/response/ServerSaveResponsePacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl\response;

use pocketcloud\cloud\network\packet\data\ServerErrorReason;
use pocketcloud\cloud\network\packet\ResponsePacket;
use pocketcloud\cloud\network\packet\util\PacketData;

final class ServerSaveResponsePacket extends ResponsePacket {

    public function __construct(private readonly ?ServerErrorReason $errorReason = null) {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeAll($this->errorReason);
    }

    public function getErrorReason(): ?ServerErrorReason {
        return $this->errorReason;
    }

    public static function create(ServerErrorReason $errorReason): self {
        return new self($errorReason);
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/network/packet/impl/response/ServerSaveResponsePacket.java
package de.pocketcloud.cloud.network.packet.impl.response;

import de.pocketcloud.cloud.network.packet.ResponsePacket;
import de.pocketcloud.cloud.network.packet.type.ActionFailureReason;
import de.pocketcloud.network.packet.data.PacketData;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Getter
public final class ServerSaveResponsePacket extends ResponsePacket {

    private ActionFailureReason errorReason;

    public ServerSaveResponsePacket(ActionFailureReason errorReason) {
        this.errorReason = errorReason;
    }

    @Override
    public void encodePayload(PacketData packetData) {
        packetData.writeAll(errorReason);
    }

    public static ServerSaveResponsePacket create(ActionFailureReason errorReason) {
        return new ServerSaveResponsePacket(errorReason);
    }
}
